server/internal/infrastructure/memory/user.go

```diff
@@ -2,165 +2,135 @@ package memory
 
 import (
 	"context"
-	"sync"
 
 	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
 	"github.com/reearth/reearth-cms/server/pkg/id"
 	"github.com/reearth/reearth-cms/server/pkg/rerror"
 	"github.com/reearth/reearth-cms/server/pkg/user"
+	"github.com/reearth/reearth-cms/server/pkg/util"
 )
 
 type User struct {
-	lock sync.Mutex
-	data map[id.UserID]*user.User
+	data util.SyncMap[id.UserID, *user.User]
```

Comment:
```suggestion
	data *util.SyncMap[id.UserID, *user.User]
```

Sorry I forgot to say. Make the data field a pointer